packages/node-app/src/services/googleAuth.service.ts

```diff
@@ -0,0 +1,60 @@
+import axios from "axios";
+import qs from 'qs';
+
+const {
+  GOOGLE_OAUTH_CLIENT_ID,
+  GOOGLE_OAUTH_CLIENT_SECRET,
+  GOOGLE_OAUTH_REDIRECT_URL
+} = process.env
+
+interface GoogleOauthToken {
+  access_token: string;
+  id_token: string;
+  expires_in: number;
+  refresh_token: string;
+  token_type: string;
+  scope: string;
+}
+
+export const getGoogleOauthToken = async (code: string, swagger: boolean):
+  Promise<{ idToken: string; accessToken: string; }> => {  
+  const options = {
+    code,
+    client_id: GOOGLE_OAUTH_CLIENT_ID,
+    client_secret: GOOGLE_OAUTH_CLIENT_SECRET,
+    redirect_uri: `${GOOGLE_OAUTH_REDIRECT_URL}${swagger ? '?swagger=true' : ''}`,
+    grant_type: 'authorization_code',
+  };
+  const { data } = await axios.post<GoogleOauthToken>(
+    "https://oauth2.googleapis.com/token",
+    qs.stringify(options),
+    { headers: { 'Content-Type': 'application/x-www-form-urlencoded' } }
+  );
+  return { idToken: data.id_token, accessToken: data.access_token };
+};
+
+interface GoogleUserResult {
+  id: string;
+  email: string;
+  verified_email: boolean;
+  name: string;
+  given_name: string;
+  family_name: string;
+  picture: string;
+  locale: string;
+}
```

Comment:
Can we have this interface at the top of the file as well, just for clarity and separation of concerns?